Sources/ARSAL_Thread.c

```diff
@@ -55,12 +80,43 @@ int ARSAL_Thread_Create(ARSAL_Thread_t *thread, ARSAL_Thread_Routine_t routine,
     if (!pthread) {
         result = -1;
     } else {
-        result = pthread_create(pthread, NULL, routine, arg);
-        if (result != 0) {
-            free(pthread);
-        } else {
-            *thread = (ARSAL_Thread_t)pthread;
+        pthread_attr_t attr;
+        pthread_attr_init(&attr);
+
+        if (!thread) {
```

Comment:
It seems to be adding a new functionality in the lib: detached thread.
A separate commit for that will be better